### PR TITLE
React & Redux group project : Space Travelers' Hub

### PR DESCRIPTION
**Pull request by ohyajk**
Hello CR, We are two people in the group in Space Traveller React group project. And we have done following steps to make this project :- 
- Use Create React App
- Use redux logger and toolkit
- Use React documentation
-  Use React components
-  Use React Router
-  Handle events in a React app
-  Write unit tests with React Testing Library

Live Demo Link : https://react-group-project-space-traveller.onrender.com/

ApprovedAppovalDeDoPleaseGIF

### CODE REVIEW

File: src/Tests/Rocket/Rockets.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+describe('testing for rockets data', () => {
+  it('test for component', () => {
+    const test2 = render(
+      <Provider store={store}>
+        <Rockets />
+      </Provider>,
+    );
+    expect(test2).toMatchSnapshot();
+  });
+});
```
Comment (TadesseDev) [inline src/Tests/Rocket/Rockets.test.js]: - [x] Dear Team, Good job adding a snapshot test, But as part of the main requirement for this milestone we need to write unit tests with React Testing Library and Jest (for unit tests). Please add more tests using `React Testing Library` and `Jest`

File: src/Pages/mission/missions_body.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const MissionsBody = () => {
+  const dispatch = useDispatch();
+  const datas = useSelector((state) => state.missions.missions);
+
+  const missions = Object.values(datas);
+
+  useEffect(() => {
+    dispatch(FetchMissions());
+  }, [dispatch]);
+
```
Comment (TadesseDev) [inline src/Pages/mission/missions_body.js]: - [x] Nice job implementing the ` dispatch(FetchMissions())` to only dispactch once the user navigate to the page, But the problem is this should happen only once. As of now if i switch to the mision back and forward i will keep hiting the API and updating the reducer. Please address this. 
- [x] This also prevent me from storing my reserved missions `as you can see from my screen record`. 
> You can check if your mission is empity before dispatching an acction to bypass this. 🙌


https://user-images.githubusercontent.com/69077061/215852778-c14cb06f-0263-409b-bee3-c3ceedea439c.mp4

File: src/Pages/mission/missions_body.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        // change the text of the button depending on the member status and action
+        let Status;
+        let Action;
+        if (mission.joined) {
+          Action = 'Leave mission';
+          Status = 'Active member';
+        } else {
+          Action = 'Join mission';
+          Status = 'NOT A MEMBER';
+        }
```
Comment (TadesseDev) [inline src/Pages/mission/missions_body.js]: - [x] Here, I can see you change the text of the button depending on the member status and action, But according to the main requirements we need two have separate components, for `join mission & Not memebr` and the other for `Leave mission & NOT A MEMBER`. Remember here we are expected to use the React conditional rendering syntax.

File: src/Pages/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      <div className="profile">
+        <div className="rockets">
+          <h2>My Rockets</h2>
+          <ul>
+            {rockets.map((rkt) => {
+              return (
+                rkt.reserved === true ? <li key={rkt.name}>{rkt.name}</li> : ''
+              );
+            })}
+          </ul>
```
Comment (TadesseDev) [inline src/Pages/Profile.jsx]: - [x] Good job rendering reserved Rockets here, But this functionality is expected be done using `filter();`. 

> you can still map once you filter. ✌️

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7

## Highlights
 - [ ] Good job overall following the template design ✅
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
- [x] Please update your [MIT](https://github.com/ohyajk/React-Group-Project/blob/Dev/MIT.md) license file with your name (both of the author of the project) and project detail, currently, it only contains `Jitender Kumar]`.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Pages/Profile.jsx (hunk 4 above)
Comment (ohyajk) [reply src/Pages/Profile.jsx]: Hello, @Tadesse-Alemayehu I have added the filter method and then mapped rockets in the Profile page accordingly.

Comment (Tadesse-Alemayehu) [reply src/Pages/Profile.jsx]: Good job @ohyajk.

Comment (ohyajk) [reply src/Pages/Profile.jsx]: Hello, @Tadesse-Alemayehu I have added the filter method and then mapped rockets in the Profile page accordingly.

File: src/Tests/Rocket/Rockets.test.js (hunk 1 above)
Comment (ohyajk) [reply src/Tests/Rocket/Rockets.test.js]: Hello, @Tadesse-Alemayehu I have added unit tests for Rockets and Profile components. I will also request my collaborator to add Mission Tests as soon as possible.

File: src/Pages/mission/missions_body.js (hunk 2 above)
Comment (Marcraphael12) [reply src/Pages/mission/missions_body.js]: ok, thanks for noticing 👌

File: src/Pages/mission/missions_body.js (hunk 3 above)
Comment (Marcraphael12) [reply src/Pages/mission/missions_body.js]: will create that👌

File: src/Pages/mission/missions_body.js (hunk 2 above)
Comment (Marcraphael12) [reply src/Pages/mission/missions_body.js]: Fixed 😎

File: src/Pages/mission/missions_body.js (hunk 3 above)
Comment (Marcraphael12) [reply src/Pages/mission/missions_body.js]: done 😎

File: src/Tests/Mission/joined_mission.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import TestRenderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { Provider } from 'react-redux';
+import store from '../../redux/configureStore';
```
Comment (Davidosky007) [inline src/Tests/Mission/joined_mission.test.js:4]: * Kindly fix these failing tests I get when I run `npm run test` 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ohyajk 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR description 👍 
* No linter errors 👍 
* Good looking readme file 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Tests/Mission/joined_mission.test.js (hunk 5 above)
Comment (ohyajk) [reply src/Tests/Mission/joined_mission.test.js:4]: Thankyou @Davidosky007 We have fixed this problem.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @ohyajk,

✔️ Passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ohyajk** [comment]
Thank you @TedLivist for the Approval. I would like to thank @Marcraphael12 for collaborating on this wonderful project. I also want to extend my gratitude to @Tadesse-Alemayehu and @Davidosky007 for giving their valuable suggestions on this project.🎉🍾🎊

**Pull request merged**